### DISCUSSION TITLE
sudo required for ot-daemon?

### DISCUSSION THREAD

**Discussion by robertcragie** [openthread/openthread#8123] [Q&A] [answered]
I am getting refamiliarized with OpenThread again after some time away.

I am using Ubuntu 22.04 ("jammy jellyfish"). When I run `ot-daemon` (not as superuser) I get an error :

```
$ ./build/posix/src/posix/ot-daemon -v 'spinel+hdlc+forkpty://build/simulation/examples/apps/ncp/ot-rcp?forkpty-arg=2'
./build/posix/src/posix/ot-daemon[156396]: Running OPENTHREAD/thread-reference-20200818-1957-g5d50088da; POSIX; Sep  2 2022 19:21:44
./build/posix/src/posix/ot-daemon[156396]: Thread version: 4
./build/posix/src/posix/ot-daemon[156396]: 52d.10:38:06.958 [C] Platform------: platformConfigureTunDevice() at netif.cpp:1622: Operation not permitted
```

If I use `sudo`, it works of course. I have tried fiddling with capabilities but I then get different errors.

I also get another permissions error using `ot-ctl`:
```
$ ./build/posix/src/posix/ot-ctl
connect session failed: Permission denied
```
Again, it works if I use `sudo`.

Do I always need to run `ot-daemon` and `ot-ctl` as superuser?

**Comment by jwhui** [comment, marked as answer]
@robertcragie , great to see you back!

OpenThread provides the Thread network interface via the Linux tun driver. By default, creating a tun interface does require superuser privileges. However, it may also be possible to provide the OpenThread process `CAP_NET_ADMIN` [capability](https://man7.org/linux/man-pages/man7/capabilities.7.html), which may allow OpenThread to run without superuser privileges.
